backend/cmd/headlamp.go

```diff
@@ -873,6 +873,14 @@ func (c *HeadlampConfig) handleClusterRequests(router *mux.Router) {
 			return
 		}
 
+		// Try to add the Authorization header if it's not already set from the client.
```

Comment:
I think instead of checking if token exists on every request this has to be solved when the context proxy is created. Like how the tls is setup to the proxy in `createProxyForContext` function.
https://github.com/headlamp-k8s/headlamp/blob/53fd76ffda97886749b1d132df4a3a73a6428c75/backend/cmd/headlamp.go#L903

Reply:
the [ReverseProxy](https://pkg.go.dev/net/http/httputil#ReverseProxy) has a `Director` function which can be used to modify the request before it is sent.

Reply:
What I meant in today's discussion, for @illume 's better context, is that if the token in the kube config is expired, then I guess Headlamp asks the user for a token. If the user then adds a token (coming from the client side), we should take that as a higher relevance than the kube config one. That's why I always verify the request in my PR.

So we should double check that if the token in the kube config fails to get access, that the frontend takes the user to the login/token-request page.